### PR TITLE
refactoring(parser): Pass around tokens instead of blocks

### PR DESCRIPTION
**Pull request by radeksimko**
Previously we were passing around (either generic `hcl` or `hclsyntax`) blocks which don't carry sufficient details about the block when it comes to completion of incomplete configs, as described in https://github.com/hashicorp/terraform-ls/issues/12#issuecomment-619157850

This PR refactors the parsing logic to pass around `hclsyntax.Tokens` instead as that carries the right amount of details we need.

The PR should however be a no-op from user's perspective.

### CODE REVIEW

File: internal/terraform/lang/hcl_parser.go [hunk 1]
```diff
@@ -10,7 +10,13 @@ import (
 
 // ParseBlock parses HCL configuration based on tfjson's SchemaBlock
 // and keeps hold of all tfjson schema details on block or attribute level
-func ParseBlock(block *hclsyntax.Block, labels []*ParsedLabel, schema *tfjson.SchemaBlock) Block {
+func ParseBlock(tokens hclsyntax.Tokens, labels []*ParsedLabel, schema *tfjson.SchemaBlock) (Block, error) {
+	hclBlock, _ := hclsyntax.ParseBlockFromTokens(tokens)
```
Comment (paultyng) [inline internal/terraform/lang/hcl_parser.go:14]: Could you add a comment here as to why you are ignoring the error?

File: internal/terraform/lang/hcl_parser.go [hunk 2]
```diff
@@ -33,6 +39,29 @@ func ParseBlock(block *hclsyntax.Block, labels []*ParsedLabel, schema *tfjson.Sc
 	return b
 }
 
+func ParseLabels(tokens hclsyntax.Tokens, schema LabelSchema) ([]*ParsedLabel, error) {
+	hclBlock, _ := hclsyntax.ParseBlockFromTokens(tokens)
```
Comment (paultyng) [inline internal/terraform/lang/hcl_parser.go:43]: Same as above, could you add a comment here?

File: internal/terraform/lang/parser.go [hunk 3]
```diff
@@ -131,26 +132,28 @@ func (bt *completableBlockType) Documentation() MarkupContent {
...
-	if block == nil {
+func (p *parser) ParseBlockFromTokens(tokens hclsyntax.Tokens) (ConfigBlock, error) {
+	if len(tokens) == 0 {
 		return nil, EmptyConfigErr
 	}
 
+	// It is probably excessive to be parsing the whole block just for type
+	// but there is no avoiding it without refactoring the upstream HCL parser
+	// and it should not hurt the performance too much
+	block, _ := hclsyntax.ParseBlockFromTokens(tokens)
```
Comment (paultyng) [inline internal/terraform/lang/parser.go:143]: Same as above, could you expand the comment here about why ignoring the error?

File: internal/terraform/lang/provider_block.go [hunk 4]
```diff
@@ -70,7 +70,9 @@ func (p *providerBlock) Labels() []*ParsedLabel {
 	if p.labels != nil {
 		return p.labels
 	}
-	p.labels = parseLabels(p.BlockType(), p.labelSchema, p.hclBlock.Labels)
+	labels, _ := ParseLabels(p.tokens, p.labelSchema)
```
Comment (paultyng) [inline internal/terraform/lang/provider_block.go:73]: Same as above, could you add a comment here?

File: internal/terraform/lang/resource_block.go [hunk 5]
```diff
@@ -81,7 +81,9 @@ func (r *resourceBlock) Labels() []*ParsedLabel {
 	if r.labels != nil {
 		return r.labels
 	}
-	r.labels = parseLabels(r.BlockType(), r.labelSchema, r.hclBlock.Labels)
+	labels, _ := ParseLabels(r.tokens, r.labelSchema)
```
Comment (paultyng) [inline internal/terraform/lang/resource_block.go:84]: Same as above, could you add a comment here?

Comment (radeksimko) [reply internal/terraform/lang/resource_block.go:84]: Given that `ParseLabels` doesn't actually need to return any error anymore I will just remove the err from the return values.

File: internal/terraform/lang/provider_block.go (hunk 4 above)
Comment (radeksimko) [reply internal/terraform/lang/provider_block.go:73]: Given that `ParseLabels` doesn't actually need to return any error anymore I will just remove the err from the return values.

**Pull request merged**